start.go

```diff
@@ -1,138 +1,27 @@
-// +build linux
-
 package main
 
 import (
-	"os"
-	"runtime"
+	"syscall"
 
 	"github.com/codegangsta/cli"
-	"github.com/coreos/go-systemd/activation"
-	"github.com/opencontainers/runc/libcontainer"
-	"github.com/opencontainers/runtime-spec/specs-go"
 )
 
-// default action is to start a container
 var startCommand = cli.Command{
 	Name:  "start",
-	Usage: "create and run a container",
+	Usage: "start signals a created container to execute the users defined process",
 	ArgsUsage: `<container-id>
 
 Where "<container-id>" is your name for the instance of the container that you
 are starting. The name you provide for the container instance must be unique on
 your host.`,
-	Description: `The start command creates an instance of a container for a bundle. The bundle
-is a directory with a specification file named "` + specConfig + `" and a root
-filesystem.
-
-The specification file includes an args parameter. The args parameter is used
-to specify command(s) that get run when the container is started. To change the
-command(s) that get executed on start, edit the args parameter of the spec. See
-"runc spec --help" for more explanation.`,
-	Flags: []cli.Flag{
-		cli.StringFlag{
-			Name:  "bundle, b",
-			Value: "",
-			Usage: `path to the root of the bundle directory, defaults to the current directory`,
-		},
-		cli.StringFlag{
-			Name:  "console",
-			Value: "",
-			Usage: "specify the pty slave path for use with the container",
-		},
-		cli.BoolFlag{
-			Name:  "detach,d",
-			Usage: "detach from the container's process",
-		},
-		cli.StringFlag{
-			Name:  "pid-file",
-			Value: "",
-			Usage: "specify the file to write the process id to",
-		},
-		cli.BoolFlag{
-			Name:  "no-subreaper",
-			Usage: "disable the use of the subreaper used to reap reparented processes",
-		},
-		cli.BoolFlag{
-			Name:  "no-pivot",
-			Usage: "do not use pivot root to jail process inside rootfs.  This should be used whenever the rootfs is on top of a ramdisk",
-		},
-	},
+	Description: `The start command signals the container to start the user's defined process.`,
 	Action: func(context *cli.Context) {
-		bundle := context.String("bundle")
-		if bundle != "" {
-			if err := os.Chdir(bundle); err != nil {
-				fatal(err)
-			}
-		}
-		spec, err := loadSpec(specConfig)
+		container, err := getContainer(context)
 		if err != nil {
 			fatal(err)
 		}
-
-		notifySocket := os.Getenv("NOTIFY_SOCKET")
-		if notifySocket != "" {
-			setupSdNotify(spec, notifySocket)
-		}
-
-		if os.Geteuid() != 0 {
-			fatalf("runc should be run as root")
-		}
-
-		status, err := startContainer(context, spec)
-		if err != nil {
+		if err := container.Signal(syscall.SIGCONT); err != nil {
```

Comment:
This probably needs at least a `container.Status()` guard like you added to `delete` in 572055d, since you only want to send SIGCONT to runtime code (and not send it after user-specified code has been executed).  That's not quite enough though, since “[the pending signal set is preserved across an execve(2)][1]”.  So “check the state and send SIGCONT if it was ‘created’” is going to be racy, and that race may result in user code seeing the extra SIGCONTs.  A more robust solution would lock a resource (setting a flag in the state registry?  Hold a Unix socket open?) when triggering code execution to avoid racing between two ‘start’ calls.

[1]: http://man7.org/linux/man-pages/man7/signal.7.html